docs/evolutionaryAlgorithm/README.md

```diff
@@ -0,0 +1,269 @@
+
+### By OD trips
+
+This fitness function requires an OD trips datasource to be available in Transition. THERE IS CURRENTLY NO WAY TO IMPORT THAT IN TRANSITION!!! See [this issue](https://github.com/chairemobilite/transition/issues/1288) for a workaround to import some.
+
+This method will take a sample of the od trips in the data source and calculate the route between each OD pair. For each pair, it will calculate a cost for this route (multiplied by the expansion factor of the odTrip, which is 1 if imported from a batch file and since it's the only way to import od trips, it's 1!). The costs for all trips will be summed and the result is the fitness of the network.
+
+In case of no result for a route calculation, the cost of a 'taxi' ride is used. The cost, fitness and non-routable functions are defined in the `defaultPreferences.config.ts` file, under `simulations` -> `geneticAlgorithms`. They should probably not be a preference, but a config somewhere, with defaults in the code itself.
+
+The goal is to minimize the cost with this method.
+
+*Note*: The odTrip routing, the way this algorithm is implemented, still uses the v1 endpoint from trRouting (the odTrip datasources has the advantage that all accessible nodes are cached for each OD pair, simplifying the calculations, and there's a lot of calculations). Since it is not supported in trRouting since commit 50418c0f, you need to use a version of trRouting prior to that. The last commit supporting v1 is c9d03a44. This algorithm is implemented in the `OdTripSimulation.ts` file in `transition-backend/src/services/simulation/methods`.
+
+### By accessibility maps
+
+This method implies to count the number of points of interest accessible by transit from different locations. This requires a datasource of places in Transition.
+
+Data for a given data source can be fetched from OpenStreetMap. Some command lines are available to facilitate the import of this data.
+
+But first, prepare the data source directory in the runtime's `imports` folder. For example, for data source named `villeray`, make sure you have the following file hierarchy:
+
+```
+<Transition runtime dir>/
+ |-- imports/
+    |-- villeray/
+       |-- polygon.geojson
+```
+
+First, a `polygon.geojson` file containing the polygon for which to import the data needs to be available in the data source directory. [geojson.io](https://geojson.io) can be used to create one.
+
+
+The following steps need to be done to prepare the data: 
+
+*Note*: all those steps can be done non-interactively. But in the absence of a proper way for tasks to announce their parameters and the lack of documentation for it, the current approach is to read to code and find out (and update this documentation!) or run it interactively.
```

Comment:
It's a bit unclear to me what you mean here when you say that they can be run non-interactively. 
Is there a way other than that running the task other than the command line ?

Reply:
No, you can only run from the command line, but there is a way for the command line not you ask you questions, by passing parameters like `--dataSourceId villeray`, but I don't remember which ones and for which tasks. I think there was one to force overwrite. (when developing and debugging those tasks, it was very handy not to spend time answering questions...)

Reply:
ah ok, I did not understand that this was what you meant!
Note*: all those steps can be run non-interactively, ie, you can pass each parameter on the command line.